### PR TITLE
Upgrade rust crates

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

**Comment by ry** [comment]
So was this done automatically by your script? Do you want to share that script?

File: build_extra/rust/BUILD.gn [hunk 1]
```diff
@@ -8,1145 +8,1786 @@
...
+    # Suppress "warning: '_addcarry_u64' is not a recognized builtin."
+    cflags = [ "-Wno-ignored-pragma-intrinsic" ]
+  }
+}
+
+## Override: don't build 'ring-test' static library.
+# static_library("ring-test") {
+#   include_dirs = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/include" ]
+#   sources = [ "$cargo_home/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/crypto/constant_time_test.c" ]
+# }
```
Comment (ry) [inline build_extra/rust/BUILD.gn]: What's this?

File: build_extra/rust/BUILD.gn [hunk 2]
```diff
@@ -8,1145 +8,1786 @@
...
-    ":slab",
-    ":string",
-    ":tokio_io",
+    ":memchr",
+    ":unicode_segmentation",
+    ":unicode_width",
+  ]
+  args = [
+    "--cap-lints",
+    "allow",
```
Comment (ry) [inline build_extra/rust/BUILD.gn]: What's "--cap-lints allow" do?

Comment (piscisaureus) [reply build_extra/rust/BUILD.gn]: Surpresses lints (warnings etc). Cargo uses this to ensure that lints introduced in a newer rust version don't break crates that were written with an older rust version in mind. 

Before, we would add '-Asomething' manually to achieve the same.

File: build_extra/rust/BUILD.gn (hunk 1 above)
Comment (piscisaureus) [reply build_extra/rust/BUILD.gn]: https://github.com/denoland/deno_third_party/blob/69028eefdc6534683eb1263be0115df679dcaed1/rust_crates/registry/src/github.com-1ecc6299db9ec823/ring-0.13.5/build.rs#L429-L448

**Comment by piscisaureus** [comment]
> you could use a wildcard for the version so we won't have to update this again?

The gitignore file is also generated. A wildcard is possible but not as trivial as it sounds, because it's not always appropriate e.g. in the case of `ring`.

**Pull request merged**